### DISCUSSION TITLE
go-sciter defined function cannot be called

### DISCUSSION THREAD

**Discussion by hutilicious** [c-smile/sciter-js-sdk#30] [Q&A] [answered]
Hi,

I am having a hard time calling a defined function w.DefineFunction in sciter.js when working with golang.
I tried several calls (document.FuncName, Window.FuncName, Window.this.FuncName,...) but nothing seems to work. Is this because go-sciter is not compatible with sciter.js (yet)?

Go:
```
w.DefineFunction("TestFunc", func(args ...*sciter.Value) *sciter.Value {
		log.Println("Called TestFunc", args)
		return sciter.NewValue("hello")
})
```

Html:
```
Window.this.TestFunc();
```

Inspector:
`TypeError: not a function`

Thanks for the hard work though, sciter.js seems very very promising.

**Comment by c-smile** [comment, marked as answer]
Please check my answer https://sciter.com/forums/topic/call_function-with-sciter-js-not-working/#post-69004

**Reply by hutilicious** [reply]
thanks for the reply. Unfortunately Inspector still shows "not a function" and the native function is not called.
Is there a pending update for the window API?

HTML:
`Window.this.call("TestFunc",'xyz')`

Inspector:
`! not a function`

Btw: Sciter.JS Version 4.4.5.9 rev. 8396

Sourcecode:
[sciterjs_golang.zip](https://github.com/c-smile/sciter-js-sdk/files/5782308/sciterjs_golang.zip)

**Reply by c-smile** [reply]
Use `Window.this.xcall("TestFunc",'xyz')`  since 4.4.5.10
See: https://github.com/c-smile/sciter-js-sdk/wiki/Window#user-content-methods

**Reply by hutilicious** [reply]
Thanks a lot Andrew! Sciter.JS really is a blessing.

**Comment by c-smile** [comment]
`window.call` is not there yet, will appear in the next build.
